ql/time/calendar.cpp

```diff
@@ -159,9 +159,14 @@ namespace QuantLib {
             Date d1 = d + n*unit;
 
             // we are sure the unit is Months or Years
-            if (endOfMonth && isEndOfMonth(d))
-                return Calendar::endOfMonth(d1);
-
+            if (endOfMonth && isEndOfMonth(d)){
```

Comment:
Should this use `Date::isEndOfMonth()` when the convention is Unadjusted?

Reply:
Yes, this is better, fixed